<?xml version='1.0'?>
<launch>

    <arg name="control_rate" default="5.0"/>
    <arg name="costmap_topic" default="/local_costmap/costmap/costmap"/>
    <arg name="goal_topic" default="/robot1/move_base_simple/goal"/>
    <arg name="localisation_topic" default="/robot1_amcl/pose"/>
    <arg name="cmd_vel_topic" default="/robot1/cmd_vel"/>
    <arg name="global_planner_service" default="/global_planner/make_plan"/>

    <node pkg="costmap_2d" name="local_costmap" type="costmap_2d_node" output="screen" >
        <rosparam command="load" ns="costmap"
                  file="$(find ropod_gazebo)/ros/config/costmap_test.yaml"/>
    </node>

    <node pkg="global_planner" name="global_planner" type="planner" output="screen" >
        <rosparam command="load" ns="costmap"
                  file="$(find ropod_gazebo)/ros/config/global_costmap_test.yaml"/>
    </node>

    <node pkg="ropod_gazebo" name="basic_navigation" type="basic_navigation" output="screen" >
        <param name="control_rate" type="double" value="$(arg control_rate)"/>
        <param name="global_planner_service" type="string" value="$(arg global_planner_service)"/>
        <remap from="~costmap" to="$(arg costmap_topic)"/>
        <remap from="~goal" to="$(arg goal_topic)"/>
        <remap from="~localisation" to="$(arg localisation_topic)"/>
        <remap from="~cmd_vel" to="$(arg cmd_vel_topic)"/>
    </node>
</launch>
